rest/route/pod_agent_test.go

```diff
@@ -301,3 +305,129 @@ func TestPodAgentNextTask(t *testing.T) {
 		})
 	}
 }
+
+func TestPodAgentEndTask(t *testing.T) {
```

Comment:
Can we test other valid cases like aborting a task or handling commit queue merge tasks?